GPU/D3D11/StencilBufferD3D11.cpp

```diff
@@ -31,42 +31,46 @@
 #define STR_HELPER(x) #x
 #define STR(x) STR_HELPER(x)
 
-struct StencilUB {
-	float u_stencilValue[4];
+struct StencilValueUB {
+	uint32_t u_stencilValue[4];
 };
 
-static const char *stencil_ps =
-"SamplerState samp : register(s0);\n"
-"Texture2D<float4> tex : register(t0);\n"
-"cbuffer base : register(b0) {\n"
-"  float4 u_stencilValue;\n"
-"};\n"
-"struct PS_IN {\n"
-"  float2 v_texcoord0 : TEXCOORD0;\n"
-"};\n"
-"float roundAndScaleTo255f(in float x) { return floor(x * 255.99); }\n"
-"float4 main(PS_IN In) : SV_Target {\n"
-"  float4 index = tex.Sample(samp, In.v_texcoord0);\n"
-"  float shifted = roundAndScaleTo255f(index.a) / roundAndScaleTo255f(u_stencilValue.x);\n"
-"  clip(fmod(floor(shifted), 2.0) - 0.99);\n"
-"  return index.aaaa;\n"
-"}\n";
-
-static const char *stencil_vs =
-"struct VS_IN {\n"
-"  float4 a_position : POSITION;\n"
-"  float2 a_texcoord0 : TEXCOORD0;\n"
-"};\n"
-"struct VS_OUT {\n"
-"  float2 v_texcoord0 : TEXCOORD0;\n"
-"  float4 position : SV_Position;\n"
-"};\n"
-"VS_OUT main(VS_IN In) {\n"
-"  VS_OUT Out;\n"
-"  Out.position = In.a_position;\n"
-"  Out.v_texcoord0 = In.a_texcoord0;\n"
-"  return Out;\n"
-"}\n";
+static const char *stencil_ps = R"(
+SamplerState samp : register(s0);
+Texture2D<float4> tex : register(t0);
+cbuffer base : register(b0) {
+  uint4 u_stencilValue;
+};
+struct PS_IN {
+  float2 v_texcoord0 : TEXCOORD0;
+};
+float4 main(PS_IN In) : SV_Target {
+  float4 index = tex.Sample(samp, In.v_texcoord0);
+	uint indexBits = uint(index.a * 256.0);
```

Comment:
Won't an alpha of `1.0` become `256` and then `indexBits & 0x100` won't work?

-[Unknown]